wordpress/wp-content/plugins/memberful-wp/src/roles.php

```diff
@@ -39,3 +49,136 @@ function memberful_wp_update_customer_roles( $old_active_role, $new_active_role,
+  $mappings = memberful_wp_get_all_plan_role_mappings();
+  return isset( $mappings[ $plan_id ] ) ? $mappings[ $plan_id ] : null;
+}
+
+/**
+ * Set the role mapping for a specific plan
+ * @param int $plan_id The plan ID
+ * @param string $role The WordPress role to assign
+ */
+function memberful_wp_set_plan_role_mapping( $plan_id, $role ) {
+  $mappings = memberful_wp_get_all_plan_role_mappings();
+  $mappings[ $plan_id ] = $role;
+  update_option( 'memberful_plan_role_mappings', $mappings );
+}
+
+/**
+ * Remove the role mapping for a specific plan
+ * @param int $plan_id The plan ID
+ */
+function memberful_wp_remove_plan_role_mapping( $plan_id ) {
+  $mappings = memberful_wp_get_all_plan_role_mappings();
+  unset( $mappings[ $plan_id ] );
+  update_option( 'memberful_plan_role_mappings', $mappings );
+}
+
+/**
+ * Get all plan role mappings
+ * @return array Array of plan_id => role mappings
+ */
+function memberful_wp_get_all_plan_role_mappings() {
+  $mappings = get_option( 'memberful_plan_role_mappings', array() );
+
+  if ( ! is_array( $mappings ) ) {
+    $mappings = array();
+  }
+
+  $allowed_roles = memberful_wp_roles_that_can_be_mapped_to();
+
+  foreach ( $mappings as $plan_id => $role ) {
+    if ( empty( $role ) || ! array_key_exists( $role, $allowed_roles ) ) {
+      $mappings[ $plan_id ] = get_option( 'default_role', 'subscriber' );
+    }
+  }
+
+  /**
+   * Filter to get all the saved per-plan role mappings.
+   *
+   * @since 1.77.0
+   *
+   * @param array $mappings The saved per-plan role mappings.
+   * @return array The role mappings.
+   */
+  return apply_filters( 'memberful_all_plan_role_mappings', $mappings );
+}
+
+/**
+ * Check if per-plan roles are enabled
+ * @return bool
+ */
+function memberful_wp_use_per_plan_roles() {
+  $use_per_plan_roles = get_option( 'memberful_use_per_plan_roles', FALSE );
+
+  /**
+   * Filter to determine if per-plan roles are enabled.
+   *
+   * @since 1.77.0
+   *
+   * @param bool $use_per_plan_roles Whether per-plan roles are enabled.
+   * @return bool Whether per-plan roles are enabled. (Default: false)
+   */
+  return apply_filters( 'memberful_use_per_plan_roles', $use_per_plan_roles );
+}
+
+/**
+ * Enable or disable per-plan roles
+ * @param bool $enabled
+ */
+function memberful_wp_set_use_per_plan_roles( $enabled ) {
+  update_option( 'memberful_use_per_plan_roles', $enabled );
+}
+
+/**
+ * Update all existing users with the new plan role mappings
+ */
+function memberful_wp_update_all_user_roles_with_plan_mappings() {
+  $mapped_users = Memberful_User_Mapping_Repository::fetch_user_ids_of_all_mapped_members();
+
+  if ( empty( $mapped_users ) ) {
+    return;
+  }
+
+  $users = get_users( array( 'fields' => 'all', 'include' => $mapped_users ) );
+
+  foreach ( $users as $user ) {
+    Memberful_Wp_User_Role_Decision::ensure_user_role_is_correct( $user );
+  }
+}
+
+/**
+ * Get the assigned role for a user.
+ *
+ * TODO: Support for multiple roles.
```

Comment:
Could we remove this line? If we've decided not to support multiple roles yet but want to add it later, we should track that elsewhere, not in code comments.